### PR TITLE
Improve stub with indentation options and better default alignments

### PR DESCRIPTION
**Pull request by rich-iannone**
This PR focuses on some first steps to improving the appearance and functionality of the gt table stub.

The first fixes (a prelude, really) involve improving the default alignments of content in the stub. With the stub serving either as row labels or indices, it's important to discern which is which and this is done as part of the auto-alignment routine in the `gt()` function.

The main thing after that is to introduce the first-class ability to set indentation on the stub elements. This is important enough to require a new function with a specialized interface for this task (`tab_stub_indent()`). 

Here is a runnable example that demonstrates one use of `tab_stub_indent()`:

```r
library(gt)
library(tidyverse)

exibble %>%
  select(-c(fctr, date, time, datetime)) %>%
  gt(rowname_col = "row", groupname_col = "group") %>%
  sub_missing() %>%
  summary_rows(
    groups = TRUE,
    columns = num,
    fns = list(
      average = "mean",
      total = "sum",
      SD = "sd"
    )
  ) %>%
  tab_options(row_group.as_column = TRUE) %>%
  tab_stub_indent(indent = 2, rows = matches("2|3|6"))
```
tab-stub-indent-example

### CODE REVIEW

**Review by cscheid — COMMENTED** [review]
Is this the kind of thing that's best done by adding a new concept ("indent") or should it be left to user CSS instead?

File: R/resolver.R [hunk 1]
```diff
@@ -311,7 +311,7 @@ translate_legacy_resolver_expr <- function(quo, null_means) {
 resolve_rows_l <- function(expr, data) {
 
   if (is_gt(data)) {
-    row_names <- dt_stub_df_get(data)$rowname
+    row_names <- dt_stub_df_get(data)$row_id
```
Comment (cscheid) [inline R/resolver.R:314]: Does this change imply breaking backwards compatibility for users of the library?

Comment (rich-iannone) [reply R/resolver.R:314]: I looked at this pretty closely and I don’t think it does. It adds better targeting of rows and doesn’t take anything away.

Comment (rich-iannone) [reply R/resolver.R:314]: We did a similar thing awhile back where we upgrading spanner column labels to have and ID and a label (previously you had to use a label for targeting) and this takes a similar approach.

Comment (cscheid) [reply R/resolver.R:314]: That sounds good; I just wanted to make sure.

**Comment by rich-iannone** [comment]
IMO it’s far better to make this first class with its own since it’s meant to establish structure and the implementation is different across output types.

**Review by cscheid — APPROVED** [review]
lgtm.

**Pull request merged**